app/config.go

```diff
@@ -240,6 +284,17 @@ func (a *App) AsymmetricSigningKey() *ecdsa.PrivateKey {
 
 func (a *App) regenerateClientConfig() {
 	a.clientConfig = utils.GenerateClientConfig(a.Config(), a.DiagnosticId(), a.License())
+
+	if _, ok := a.clientConfig["CustomServiceTermsId"]; ok {
+		if result := <-a.Srv.Store.ServiceTerms().Get(true); result.Err != nil && result.Err.Id != ERROR_SERVICE_TERMS_NO_ROWS_FOUND {
+			panic(result.Err)
```

Comment:
I'm not sure a panic here is appropriate. Perhaps on startup when validating the config, but I'm inclined to make the behaviour here more graceful in the presence of an error than to take down the entire server.